Data_Structures_Questions.md

```diff
@@ -4,50 +4,100 @@ Answer the following questions for each of the data structures you implemented a
 
 1. What is the runtime complexity of `enqueue`?
 
+    O(1) because it only ever has to do one thing and doesn't have to loop over anything
+
 2. What is the runtime complexity of `dequeue`?
 
+    O(1) because it only ever has to do one thing and doesn't have to loop over anything
+
 3. What is the runtime complexity of `len`?
 
+    O(1) because it only ever has to do one thing and doesn't have to loop over anything
+
 ## Binary Search Tree
 
-1. What is the runtime complexity of `insert`? 
+1. What is the runtime complexity of `insert`?
+
+    O( log n) because it uses recursion and binary search is always log n
 
 2. What is the runtime complexity of `contains`?
 
-3. What is the runtime complexity of `get_max`? 
+    O( log n) because it uses recursion and binary search is always log n
+
+3. What is the runtime complexity of `get_max`?
+
+    O(log n) because it has to iterate to find the max and uses recursion
 
 ## Heap
 
 1. What is the runtime complexity of `_bubble_up`?
 
+    O(log n) because it uses a while loop(n) then an if statement(1)
+
 2. What is the runtime complexity of `_sift_down`?
 
+    O(log n) because it uses a while loop(n) then an if statement(1)
+
```

Comment:
For 3 and 4 here your answer is correct, but we can just say `O(log n)` since it is the overarching complexity, the `+ 1` in this case is negligible.